qabelbox/src/main/java/de/qabel/desktop/repository/sqlite/SqliteContactRepository.java

```diff
@@ -195,4 +204,71 @@ public synchronized Contact findByKeyId(Identity identity, String keyId) throws
             throw new EntityNotFoundExcepion("exception while searching contact: " + e.getMessage(), e);
         }
     }
+
+    @Override
+    public Collection<Pair<Contact, List<Identity>>> find(Identities identities, String searchString) throws PersistenceException {
+        try {
+
+            Map<String, String> identityKeyMap = new HashMap<>();
+            for (Identity identity : identities.getIdentities()) {
+                String key = "id_" + identity.getId();
+                identityKeyMap.put(key, identity.getKeyIdentifier());
+            }
+
+            ContactIdentitiesHydrator customHydrator = new ContactIdentitiesHydrator(
```

Comment:
You could also constructor-inject this (which needs some refactoring, but ultimatively we like to use dependency injection)